src/shared/components/oncoprint/OncoprintUtils.ts

```diff
@@ -1130,113 +1153,223 @@ export function makeHeatmapTracksMobxPromise(
     sampleMode: boolean
 ) {
     return remoteData<IHeatmapTrackSpec[]>({
-        await: () => [
-            oncoprint.props.store.filteredSamples,
-            oncoprint.props.store.filteredPatients,
-            oncoprint.props.store.molecularProfileIdToMolecularProfile,
-            oncoprint.props.store.geneMolecularDataCache,
-        ],
-        invoke: async () => {
+        await: () => {
             const molecularProfileIdToMolecularProfile = oncoprint.props.store
                 .molecularProfileIdToMolecularProfile.result!;
             const molecularProfileIdToAdditionalTracks =
                 oncoprint.molecularProfileIdToAdditionalTracks;
 
-            const geneProfiles = _.filter(
-                _.values(molecularProfileIdToAdditionalTracks),
-                d =>
-                    d.molecularAlterationType !==
-                    AlterationTypeConstants.GENERIC_ASSAY
+            const cacheQueries = getGeneProfileQueries(
+                molecularProfileIdToAdditionalTracks,
+                oncoprint.props.store.geneCache
             );
+
+            const mutationQueries = cacheQueries.filter(query => {
+                const profileType =
+                    molecularProfileIdToMolecularProfile[
+                        query.molecularProfileId
+                    ]?.molecularAlterationType;
+                return (
+                    profileType === AlterationTypeConstants.MUTATION_EXTENDED
+                );
+            });
+
+            return [
+                oncoprint.props.store.filteredSamples,
+                oncoprint.props.store.filteredPatients,
+                oncoprint.props.store.molecularProfileIdToMolecularProfile,
+                oncoprint.props.store.geneMolecularDataCache,
+                ...mutationQueries.map(query =>
+                    oncoprint.props.store.annotatedMutationCache.get({
+                        entrezGeneId: query.entrezGeneId!,
+                    })
+                ),
+            ];
+        },
+        invoke: async () => {
```

Comment:
This invoke function now got too long. We should probably try dividing it into sub functions.